x/ibc-transfer/types/msgs.go

```diff
@@ -39,26 +41,33 @@ func (MsgTransfer) Type() string {
 }
 
 // ValidateBasic performs a basic check of the MsgTransfer fields.
-// NOTE: timeout height and timestamp values can be 0 to disable the timeout.
+// NOTE: timeout height or timestamp values can be 0 to disable the timeout.
 func (msg MsgTransfer) ValidateBasic() error {
 	if err := host.PortIdentifierValidator(msg.SourcePort); err != nil {
 		return sdkerrors.Wrap(err, "invalid source port ID")
 	}
 	if err := host.ChannelIdentifierValidator(msg.SourceChannel); err != nil {
 		return sdkerrors.Wrap(err, "invalid source channel ID")
 	}
-	if !msg.Amount.IsAllPositive() {
-		return sdkerrors.Wrap(sdkerrors.ErrInsufficientFunds, msg.Amount.String())
-	}
 	if !msg.Amount.IsValid() {
 		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, msg.Amount.String())
 	}
+	if !msg.Amount.IsPositive() {
+		return sdkerrors.Wrap(sdkerrors.ErrInsufficientFunds, msg.Amount.String())
+	}
 	if msg.Sender.Empty() {
 		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, "missing sender address")
 	}
 	if msg.Receiver == "" {
 		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, "missing recipient address")
 	}
+
+	// sanity check that validate basic on fungible token packet passes
+	// NOTE: this should always pass since validation checks should be the same
+	packet := NewFungibleTokenPacketData(msg.Amount, msg.Sender.String(), msg.Receiver, msg.Source)
+	if err := packet.ValidateBasic(); err != nil {
+		return sdkerrors.Wrap(err, "unexpected failure in fungible token packet validation, please open an issue")
```

Comment:
nit, I'd just return the error tbh

Reply:
yea, I'll just move the error message to the comments. I'd still like to indicate that if that error occurs, it is a bug and it should be addressed/fixed